### PR TITLE
feat: add @nodevu/fetchindex

### PR DESCRIPTION
**Pull request by bnb**
this PR abstracts the `util/prod/version.js` file from `@nodevu/core` into a new package, `@nodevu/fetchindex`. This is the result of a discussion with @ljharb and @wesleytodd at the Node.js Collaborator Summit in Dublin, Ireland.

### CODE REVIEW

**Comment by bnb** [comment]
worth noting, this API will probably immediately change as I spin out `optionsParser` next

**Comment by ljharb** [comment]
The package doesn't seem to be published, ftr

File: fetchindex/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		"url": "https://github.com/cutenode/nodevu/issues"
+	},
+	"homepage": "https://github.com/cutenode/nodevu#readme",
+	"devDependencies": {
+		"@biomejs/biome": "1.9.4",
+		"c8": "^10.1.2",
+		"luxon": "^3.5.0",
+		"undici": "^6.20.1"
+	}
+}
```
Comment (ljharb) [inline fetchindex/package.json:32]: it'd be great to add `engines.node` here as well :-)

**Comment by wesleytodd** [comment]
Sorry this got lost in my notifications, but consider this an after merge 👍 

EDIT: I miss tabs in projects. This made me happy today.

**Pull request merged**